api/tools/src/main/java/org/pmiops/workbench/tools/CreateExtractionBillingProjectWorkspace.java

```diff
@@ -0,0 +1,133 @@
+package org.pmiops.workbench.tools;
+
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.firecloud.api.BillingApi;
+import org.pmiops.workbench.firecloud.model.*;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Sets up workspace for Extraction service account 1. Creates Terra BP 2. Create Terra Workspace 3.
+ * Share Workspace with given users
+ */
+@Configuration
+public class CreateExtractionBillingProjectWorkspace {
+
+  private static Option billingAccountOpt =
+      Option.builder().longOpt("billing-account").required().hasArg().build();
+
+  private static Option billingProjectNameOpt =
+      Option.builder().longOpt("billing-project-name").required().hasArg().build();
+
+  private static Option workspaceNameOpt =
+      Option.builder().longOpt("workspace-name").required().hasArg().build();
+
+  private static Option ownersOpt = Option.builder().longOpt("owners").required().hasArg().build();
+
+  private static Options options =
+      new Options()
+          .addOption(billingAccountOpt)
+          .addOption(billingProjectNameOpt)
+          .addOption(workspaceNameOpt)
+          .addOption(ownersOpt);
+
+  private static final Logger log =
+      Logger.getLogger(CreateExtractionBillingProjectWorkspace.class.getName());
+
+  @Bean
+  ImpersonatedServiceAccountApiClientFactory apiClientFactory(WorkbenchConfig config) {
+    // TODO : replace string with `config.auth.extractionServiceAccount` once that value is in the
```

Comment:
Should this just be a flag (which can be supplied from the config via devstart)? Doesn't seem correct to me that a command line tool might try to read the live config value from the DB.